azext_edge/edge/providers/rpsaas/adr2/command.md

```diff
@@ -0,0 +1,476 @@
+swagger:
+https://github.com/Azure/azure-rest-api-specs-pr/blob/RPSaaSMaster/specification/deviceregistry/resource-manager/Microsoft.DeviceRegistry/preview/2024-07-01-preview/deviceregistry.json
+
+## Assets
+id: `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.DeviceRegistry/assets/my-asset`
+
+discovered assets are a different thing but have the same props: `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.DeviceRegistry/discoveredAssets/my-asset`
+- the discovered asset is supposed to be created from the edge/cluster rather than cloud
+- create/update/delete are still allowed via the API
+- list via the regular vs discovered should yeild different asset lists
+
+
+Main changes:
+- datapoints are now under dataset
+- addition of topic property
+- removal of some other properties
+- somehow event/dataset requirement is no longer there?
+
+### asset
+
+  create
+
+	--asset [or --name]
+	--instance
+	--resource-group
+	--endpoint-profile
+	[optional]
+	--custom-attribute
+	--data-set-file
+	--default-topic-path
+	--default-topic-retention (`keep`, `never`) [may turn this into a bool flag]
+	--description
+	--disabled [flag]
+	--display-name
+	--documentation-uri
+	--event [key=value pairs, `event_notifier`, `name` required]
+	--event-file
+	--external-asset-id
+	--hardware-revision
+	--instance-resource-group
+	--instance-subscription
+	--location
+	--manufacturer
+	--manufacturer-uri
+	--model
+	--product-code
+	--serial-number
+	--software-revision
+	--dataset-publishing-interval
+	--dataset-sampling-interval
+	--dataset-queue-size
+	--event-publishing-interval
+	--event-sampling-interval
+	--event-queue-size
+	--tags
+
+  delete
+
+	--asset [or --name]
+	--resource-group
+
+  list
+
+	[optional]
+	--discovered [bool flag]
+	--resource-group
+	--endpoint-profile
+	--instance
+	--instance-resource-group [require for instance]
+	--display-name
+	--documentation-uri
+	--external-asset-id
+	--hardware-revision
+	--instance-resource-group
```

Comment:
What is the short of this  --ig or --irg ? I kinda lean --ig because --resource-group is -g.

Reply:
I guess it will be --ig then